### PR TITLE
add --reserved-cpus kubelet command option

### PR DESCRIPTION
**Pull request by mburke5678**
https://issues.redhat.com/browse/OSDOCS-965
https://issues.redhat.com/browse/OCPNODE-217

### CODE REVIEW

**Comment by mburke5678** [comment]
@rphillips Can you find someone to help with this PR? 
Has the NODE issue been through QE?
Can the user add the parameter directly to the KubeletConfig YAML or must use `c set env KubeletConfig/<name> reserved-cpus=0,1,2,3`
Does using this parameter affect any of the other sections in https://docs.openshift.com/container-platform/4.4/nodes/nodes/nodes-nodes-resources-configuring.html

**Comment by mburke5678** [comment]
@rphillips PTAL

**Comment by rphillips** [comment]
@mburke5678 this option gets set by a KubeletConfig custom resource


```
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  name: set-reserved-cpus  
spec:
  reservedSystemCPUs: "0,1,2,3"
  machineConfigPoolSelector:
    matchLabels:
      pools.operator.machineconfiguration.openshift.io/worker: ""
```

Setting the option would be similar to the following documentation: https://docs.openshift.com/container-platform/4.6/nodes/nodes/nodes-nodes-managing-max-pods.html

**Comment by mburke5678** [comment]
@rphillips I added the KubeletConfig CR that you suggested. Can you please take another look?

File: modules/nodes-nodes-resources-cpus-reserve.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-nodes-resources-cpus
+
+[id="nodes-nodes-resources-cpus-reserve_{context}"]
+= Reserving CPUs for nodes
+
+To explicitly define a list of CPUs that are reserved for control plane nodes or worker nodes, create a `KubeletConfig` custom resource (CR) to define the `reserved-cpus` parameter. This list supersedes the CPU counts that might be specified using the `--system-reserved` and `--kube-reserved` parameters.
```
Comment (rphillips) [inline modules/nodes-nodes-resources-cpus-reserve.adoc:8]: systemReserved and kubeReserved options.

File: nodes/nodes/nodes-nodes-resources-cpus.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// modules required to cover the user story. You can also include other
+// assemblies.
+
+include::modules/nodes-nodes-resources-cpus-reserve.adoc[leveloffset=+1]
+
+.Additional resources
+
+* For information on the static CPU manager policy, see xref:../../scalability_and_performance/using-cpu-manager.adoc[Using CPU Manager].
+
+* For more information on the `--system-reserved` and `--kube-reserved` parameters, see xref:../../nodes/nodes/nodes-nodes-resources-configuring.html#nodes-nodes-resources-configuring-about_nodes-nodes-resources-configuring[Allocating resources for nodes in an {product-title} cluster].
```
Comment (rphillips) [inline nodes/nodes/nodes-nodes-resources-cpus.adoc:22]: systemReserved and kubeReserved options here too

**Comment by mburke5678** [comment]
@sunilcio Is this a PR that you can review for QE? Thank you in advance!

**Comment by sunilcio** [comment]
@mburke5678 missed this one earlier. LGTM, thanks!

**Comment by mburke5678** [comment]
Preview: https://deploy-preview-23317--osdocs.netlify.app/openshift-enterprise/latest/nodes/nodes/nodes-nodes-resources-cpus.html

File: nodes/nodes/nodes-nodes-resources-cpus.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+:context: nodes-nodes-resources-cpus
+[id="nodes-nodes-resources-cpus"]
+= Allocating specific CPUs for nodes in a cluster
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+
+When using the xref:../../scalability_and_performance/using-cpu-manager.adoc[static CPU Manager policy], you can reserve specific CPUs for use by specific nodes in your cluster. For example, on a system with 24 CPUs, you could reserve CPUs numbered 0 - 3 for the control plane allowing the workers to use CPUs 4 - 23.
```
Comment (codyhoag) [inline nodes/nodes/nodes-nodes-resources-cpus.adoc:9]: s/for the control plane allowing/for the control plane, allowing

Is this an instance where we could replace `workers` with `compute nodes`?

File: modules/nodes-nodes-resources-cpus-reserve.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+----
+<1> Specify a name for the CR.
+<2> Specify the core IDs of the CPUs you want to reserve for the nodes associated with the MCP.
+<3> Specify the label from the MCP.
+
+. Create the CR object:
++
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
```
Comment (codyhoag) [inline modules/nodes-nodes-resources-cpus-reserve.adoc:63]: s/\<file-name\>/<file_name>

**Review by codyhoag — APPROVED** [review]
Just a few nits; looks great otherwise! Be sure to apply the appropriate labels before merging :slightly_smiling_face:

**Pull request merged**